stacker/hooks/keypair.py

```diff
@@ -2,21 +2,183 @@
+        return None
+
+    keypair = import_key_pair(ec2, keypair_name, public_key_data)
+    return {
+        "status": "imported",
+        "key_name": keypair["KeyName"],
+        "fingerprint": keypair["KeyFingerprint"],
+    }
+
+
+def create_key_pair_in_ssm(ec2, ssm, keypair_name, parameter_name,
+                           kms_key_id=None):
+    keypair = create_key_pair(ec2, keypair_name)
+    try:
+        kms_key_label = 'default'
+        kms_args = {}
+        if kms_key_id:
+            kms_key_label = kms_key_id
+            kms_args = {"KeyId": kms_key_id}
+
+        logger.info("Storing generated key in SSM parameter \"%s\" "
+                    "using KMS key \"%s\"", parameter_name, kms_key_label)
+
+        ssm.put_parameter(
+            Name=parameter_name,
+            Description="SSH private key for KeyPair \"{}\" "
+                        "(generated by Stacker)".format(keypair_name),
+            Value=keypair["KeyMaterial"],
+            Type="SecureString",
+            Overwrite=False,
+            **kms_args)
+    except ClientError:
+        # Erase the key pair if we failed to store it in SSM, since the
+        # private key will be lost anyway
+
+        logger.exception("Failed to store generated key in SSM, deleting "
+                         "created key pair as private key will be lost")
+        ec2.delete_key_pair(KeyName=keypair_name, DryRun=False)
+        return None
+
+    return {
+        "status": "created",
+        "key_name": keypair["KeyName"],
+        "fingerprint": keypair["KeyFingerprint"],
+    }
+
+
+def create_key_pair(ec2, keypair_name):
+    keypair = ec2.create_key_pair(KeyName=keypair_name, DryRun=False)
+    logger.info(KEYPAIR_LOG_MESSAGE,
+                keypair["KeyName"],
+                keypair["KeyFingerprint"],
+                "created")
+    return keypair
+
+
+def create_key_pair_local(ec2, keypair_name, dest_dir):
+    dest_dir = utils.full_path(dest_dir)
+    if not os.path.isdir(dest_dir):
+        logger.error("\"%s\" is not a valid directory", dest_dir)
+        return None
+
+    file_name = "{0}.pem".format(keypair_name)
+    key_path = os.path.join(dest_dir, file_name)
+    if os.path.isfile(key_path):
+        # This mimics the old boto2 keypair.save error
+        logger.error("\"%s\" already exists in \"%s\" directory",
+                     file_name, dest_dir)
+        return None
+
+    # Open the file before creating the key pair to catch errors early
+    with open(key_path, "wb") as f:
+        keypair = create_key_pair(ec2, keypair_name)
+        f.write(keypair["KeyMaterial"].encode("ascii"))
+
+    return {
+        "status": "created",
+        "key_name": keypair["KeyName"],
+        "fingerprint": keypair["KeyFingerprint"],
+        "file_path": key_path
+    }
+
+
+def interactive_prompt(keypair_name, ):
+    if not sys.stdin.isatty():
+        return None, None
+
+    try:
+        while True:
+            action = input(
```

Comment:
worth using this input method since we use it most other places? https://github.com/cloudtools/stacker/blob/1cb24aa1a596261d533a5dfc4536757a3743b6ab/stacker/ui.py#L14

Reply:
Done!